src/util/schnorr.rs

```diff
@@ -17,3 +17,64 @@
 //!
 
 pub use secp256k1::schnorrsig::{PublicKey, KeyPair};
+use secp256k1::{Secp256k1, Verification};
+use hashes::{Hash, HashEngine};
+use util::taproot::{TapBranchHash, TapTweakHash};
+
+/// Untweaked Schnorr public key
+pub struct UntweakedPublicKey(pub PublicKey);
+
+/// Tweaked Schnorr public key
+pub struct TweakedPublicKey(PublicKey);
+
+impl UntweakedPublicKey {
+    /// Tweaks an untweaked public key given an untweaked key and optional script tree merkle root.
+    /// 
+    /// This is done by using the equation Q = P + H(P|c)G, where 
+    ///  * Q is the tweaked key
+    ///  * P is the internal key
+    ///  * H is the hash function
+    ///  * c is the commitment data
+    ///  * G is the generator point
+    pub fn tweak<C: Verification>(&self, secp: Secp256k1<C>, merkle_root: Option<TapBranchHash>) -> TweakedPublicKey {
+        //PR #677 has a function `TapTweakHash::from_key_and_tweak(..)` which can be simpler to use here.
+        
+        //Commitment data.
+        let h: Vec<u8> = match merkle_root {
+            Some(hash) => hash.to_vec(),
+            None => vec![]
+        };
```

Comment:
This looks incorrect since it will lead to an non-self-tweaked public key is the script is absent, which is against BIP-341 recommendations

Reply:
Sorry,  I am not getting where the cited code block exists in this PR